Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1202,48 +1202,75 @@ namespace Babylon
             XRRay(const Napi::CallbackInfo& info)
                 : Napi::ObjectWrap<XRRay>{info}
             {
-                bool originSet = false;
-                bool directionSet = false;
-                bool matrixSet = false;
-                if (info[0].IsObject())
+                auto argLength = info.Length();
+                xr::Ray tempVals{};
+                tempVals.Direction.Z = -1.0;
+
+                m_origin = Napi::Persistent(Napi::Object::New(info.Env()));
+                m_direction = Napi::Persistent(Napi::Object::New(info.Env()));
+                m_matrix = Napi::Persistent(Napi::Float32Array::New(info.Env(), MATRIX_SIZE));
+
+                // The constructor is either sent a BABYLON.Vector3, {}, an XRRigidTransform, or {x,y,z,w},{x,y,z,w}
```

Comment:
Is it expected that the `w` component is ignored completely?

Reply:
I've updated the code to throw an error if the w component isn't what is expected, as per xrray docs